### PR TITLE
Updates to User Testing and Index page

### PR DESCRIPTION
**Pull request by junwei-tan**

### CODE REVIEW

File: src/main/asciidoc/testing-guide.adoc [hunk 1]
```diff
@@ -96,7 +95,7 @@ https://forms.gle/mv4PmppxTaHfbwBq6[https://forms.gle/mv4PmppxTaHfbwBq6]
...
+Testing was done with a  Google Form which systematically guides the user through each feature of WeMeet while collecting their comments. As WeMeet is meant to be a general scheduling platform, we will be doing our user testing on the general public. Emails and text messages with the user testing link were sent to friends and family. We will be conducting one round of internal testing followed by 3 rounds of user testing. As testing rounds occur, we will also be refining the features of WeMeet and making small additions to the user testing form to include them. Email addresses of respondents are collected to ensure all responses collected are legitimate.
```
Comment (xumarcus) [inline src/main/asciidoc/testing-guide.adoc:98]: `  Google Form` double space

Comment (xumarcus) [inline src/main/asciidoc/testing-guide.adoc:98]:
```suggestion
Testing was done with aGoogle Form which systematically guides the user through each feature of WeMeet while collecting their comments. As WeMeet is meant to be a general scheduling platform, we will be doing our user testing on the general public. Emails and text messages with the user testing link were sent to friends and family. We will be conducting one round of internal testing followed by 3 rounds of user testing. As testing rounds occur, we will also be refining the features of WeMeet and making small additions to the user testing form to include them. Email addresses of respondents are collected to ensure all responses collected are legitimate.
```

File: src/main/asciidoc/testing-guide.adoc [hunk 2]
```diff
@@ -117,6 +116,46 @@ This approach of a guided Google Form allows us to systematically test every fea
...
 11th - 14th July: First round  +
+14th - 15th July: Review first round feedback +
 15th - 18th July: Second round  +
+18th - 19th July: Review second round feedback +
 19th - 22nd July: Third round
-Analysing, adapting to feedback will happen as a manner of continuous deployment. This approach takes inspiration from the Design Thinking Methodology : Create, Test, Improve, Iterate.
\ No newline at end of file
+22nd - 24th July: Review third round feedback
+
+Analysing, adapting to feedback will happen as a manner of continuous deployment. This approach takes inspiration from the Design Thinking Methodology : Create, Test, Improve, Iterate.
```
Comment (xumarcus) [inline src/main/asciidoc/testing-guide.adoc:125]:
```suggestion
We continuously analyze and adapt to feedback, taking inspiration from the Design Thinking Methodology : Create, Test, Improve, Iterate.
```

Comment (xumarcus) [reply src/main/asciidoc/testing-guide.adoc:125]: We don't have CI/CD

File: src/main/asciidoc/testing-guide.adoc [hunk 3]
```diff
@@ -117,6 +116,46 @@ This approach of a guided Google Form allows us to systematically test every fea
...
+
+The third round of user testing was conducted from 19th to 22nd July with 5 responses.
+|===
+|*Section*|*Feedback Received*|*Action Taken*
+
+|Scheduler functionality|The scheduler lacks more advanced features such as calendar integration.|We will include more advanced features in future updates.
+|===
+
+
+== System testing
```
Comment (xumarcus) [inline src/main/asciidoc/testing-guide.adoc:158]: Describe the details of the tests, like what's exactly tested

File: src/main/asciidoc/testing-guide.adoc [hunk 4]
```diff
@@ -86,8 +86,7 @@ We tested out the following API for backend.
 ** After solver terminates, when client `GET`, then server responds with `200 OK` with `solved` set to `true` and response contains valid solution.
 * Given valid request, when client `POST /api/rosterPlan/publish`, then server responds with `200 OK` and overrides original `rosterPlan`.
 
-== System testing
-TODO
+
```
Comment (xumarcus) [inline src/main/asciidoc/testing-guide.adoc:89]: Remove extra blank lines

File: src/main/asciidoc/testing-guide.adoc [hunk 5]
```diff
@@ -86,8 +86,7 @@ We tested out the following API for backend.
...
 
 == User testing
 === Google Form Link for User Testing
```
Comment (xumarcus) [inline src/main/asciidoc/testing-guide.adoc:92]: ?

**Comment by xumarcus** [comment]
Please have more descriptive branch names

File: src/main/asciidoc/testing-guide.adoc [hunk 6]
```diff
@@ -117,6 +116,46 @@ This approach of a guided Google Form allows us to systematically test every fea
...
+|*Section*|*Feedback Received*|*Action Taken*
+
+|Scheduler functionality|The scheduler lacks more advanced features such as calendar integration.|We will include more advanced features in future updates.
+|===
+
+
+== System testing
+System testing was conducted manually after each major revision (e.g. before each round of user testing). A developer will use
+the user testing questions as a guide to test each and every functionality of the webapp to ensure that
+it is functioning as expected.
```
Comment (junwei-tan) [inline src/main/asciidoc/testing-guide.adoc:161]: test

Comment (junwei-tan) [inline src/main/asciidoc/testing-guide.adoc:161]:
```suggestion
it is functioning as expected.
During the process, general webpage navigation, account authentication and scheduler functionality is tested. More specifically, we follow the user testing google forms which have detailed tasks such as creating a account, logging into an account, creating an event etc.
```

File: src/main/asciidoc/testing-guide.adoc [hunk 7]
```diff
@@ -117,6 +116,47 @@ This approach of a guided Google Form allows us to systematically test every fea
...
+During the process, general webpage navigation, account authentication and scheduler functionality is tested. More specifically, we follow the user testing google forms which have detailed tasks such as creating a account, logging into an account, creating an event etc.
```
Comment (xumarcus) [inline src/main/asciidoc/testing-guide.adoc:162]: As in exactly step by step description of how you test and what results you get
Just have a few scenarios, test it, document, and done

Comment (junwei-tan) [inline src/main/asciidoc/testing-guide.adoc:162]:
```suggestion
|===
||Date Tested||

||10th July|15th July|19th July
|I am able to sign up for an account|ok|ok|ok
|I am able to sign in to my account|ok|ok|ok
|I am able to sign out of my account|ok|ok|ok
|I receive an error message when i sign in with a wrong email|ok|ok|ok
|I receive an error message when i sign in with a wrong password|ok|ok|ok
|I can create a new meeting with my desired meeting name|ok|ok|ok
|I am able to click into the meeting’s page|ok|ok|ok
|As a host, I am able to create time-slots|ok|ok|ok
|As a host, I am able to add capacity for individual time slots|ok|ok|ok
|As a host, I am able to add respondent’s emails|ok|ok|ok
|As a respondent, I am able to rank time slots|ok|ok|ok
|As a host, I am able to generate a roster|ok|ok|ok
|As a respondent, I can view roster generated by host|ok|ok|ok
|===
```

**Pull request merged**